BUILD

```diff
@@ -53,171 +59,147 @@ filegroup(
         "test/io.tcl",
         ":util",
     ],
-    visibility = [":__subpackages__"],
 )
 
 filegroup(
     name = "constraints-sram",
     srcs = [
         "test/constraints-sram.sdc",
+    ],
+    data = [
         ":util",
     ],
     visibility = [":__subpackages__"],
 )
 
-build_openroad(
-    name = "tag_array_64x184",
-    abstract_stage = "floorplan",
-    io_constraints = ":io-sram",
-    sdc_constraints = ":constraints-sram",
-    stage_args = {
-        "floorplan": [
-            "CORE_UTILIZATION=40",
-            "CORE_ASPECT_RATIO=2",
-        ],
-        "place": ["PLACE_DENSITY=0.65"],
+synth(
+    name = "tag_array_synth",
+    arguments = {
+        "SDC_FILE": "$(location :constraints-sram)",
     },
-    verilog_files = ["//another:tag_array_64x184.sv"],
-    visibility = [":__subpackages__"],
+    data = [
+        ":constraints-sram",
+    ],
+    module_top = "tag_array_64x184",
+    verilog_files = ["test/mock/tag_array_64x184.sv"],
+)
+
+SRAM_FLOOR_PLACE_ARGUMENTS = {
+    "IO_CONSTRAINTS": "$(location :io-sram)",
+}
+
+floorplan(
+    name = "tag_array_floor",
+    src = ":tag_array_synth",
+    arguments = SRAM_FLOOR_PLACE_ARGUMENTS | {
+        "CORE_UTILIZATION": "40",
+        "CORE_ASPECT_RATIO": "2",
+    },
+    data = [
+        ":io-sram",
+    ],
 )
 
-build_openroad(
-    name = "lb_32x128",
-    abstract_stage = "floorplan",
-    io_constraints = ":io-sram",
-    mock_area = 1,
-    sdc_constraints = ":constraints-sram",
-    stage_args = {
-        "floorplan": [
-            "CORE_UTILIZATION=40",
-            "CORE_ASPECT_RATIO=2",
-        ],
-        "place": ["PLACE_DENSITY=0.65"],
+place(
+    name = "tag_array_place",
+    src = ":tag_array_floor",
+    arguments = SRAM_FLOOR_PLACE_ARGUMENTS | {
+        "PLACE_DENSITY": "0.65",
     },
+    data = [
+        ":io-sram",
+    ],
+)
+
+abstract(
+    name = "tag_array_abstract",
+    src = ":tag_array_place",
+)
+
+synth(
+    name = "lb_synth",
+    arguments = {
+        "SDC_FILE": "$(location :constraints-sram)",
+    },
+    data = [
+        ":constraints-sram",
+    ],
+    module_top = "lb_32x128",
     verilog_files = ["test/rtl/lb_32x128.sv"],
 )
 
-build_openroad(
```

Comment:
Can we still support the build_openroad() macro and have it expand to these new rules?

Reply:
Make a naming convention through the build_openroad() macro?

Reply:
Yes we can!